### PR TITLE
Tutor filtering

### PR DESCRIPTION
**Pull request by nareddyt**
Functions for filtering tutors

Also added a save user function

### CODE REVIEW

**Comment by nareddyt** [comment]
Github is showing a weird diff. The only changes are that I added 2 new functions to the dao, did not delete any other code.

File: api/dao/user_dao.js [hunk 1]
```diff
@@ -97,41 +91,96 @@ module.exports = {
...
-                // Look for students with the same username
-                Student.find({_id: username}, function (err, docs) {
-                    if (err) {
-                        console.error('Error checking if username is taken:', err);
-                        callback(err);
+    /**
+     * TODO: Availability is not used currently
+     * TODO: No grade level checking for subject
+     */
+    getAllAvailableTutors: function(subject, availability, callback) {
```
Comment (ganhari123) [inline api/dao/user_dao.js]: So i am a little confused, why do we have a function with functions in them?

Comment (nareddyt) [reply api/dao/user_dao.js]: Accessing mongodb is async, we can't do it synchronously. You pass in a callback that gets called when the tutors are all retrieved from mongo. So you would do something like this:

```javascript
function main() {
dao.getAllAvailableTutors('CS', 'Tuesdays', printTutors);
}

function printTutors(err, tutors) {
  console.log(tutors);
}```

Comment (nareddyt) [reply api/dao/user_dao.js]: The functions in the function are just boolean functions used for filtering. They're used by the ```tutors.filter``` lines near the end of the main function. You can have functions inside functions in javascript

**Pull request merged**